modules/component-store/spec/component-store.spec.ts

```diff
@@ -638,6 +595,69 @@ describe('Component Store', () => {
     );
   });
 
+  describe('throws an error', () => {
+    it('when synchronous error is thrown within updater', () => {
+      const componentStore = new ComponentStore({});
+      const error = new Error('ERROR!');
+      const updater = componentStore.updater(() => {
+        throw error;
+      });
+
+      expect(() => updater()).toThrow(error);
+    });
+
+    it('when synchronous error is thrown within setState callback', () => {
+      const componentStore = new ComponentStore({});
+      const error = new Error('ERROR!');
+
+      expect(() => {
+        componentStore.setState(() => {
+          throw error;
+        });
+      }).toThrow(error);
+    });
+
+    it('when synchronous error is thrown within patchState callback', () => {
+      const componentStore = new ComponentStore({});
+      const error = new Error('ERROR!');
+
+      expect(() => {
+        componentStore.patchState(() => {
+          throw error;
+        });
+      }).toThrow(error);
+    });
+
+    it(
+      'when asynchronous observable throws an error with updater',
+      marbles((m) => {
+        const componentStore = new ComponentStore({});
+        const error = new Error('ERROR!');
+        const updater = componentStore.updater<unknown>(() => ({}));
+        const asyncObs$ = m.cold('-#', {}, error);
+
+        expect(() => {
+          updater(asyncObs$);
+          m.flush();
+        }).toThrow(error);
+      })
+    );
+
+    it(
+      'when asynchronous observable throws an error with patchState',
```

Comment:
Does this test `isSyncUpdate` to be `false`?

Shouldn't the validation be that it **does not throw** then? And once flushed shouldn't `throwError(() => error)` happen?

Reply:
Yes, you're right. I'll update the test to make sure the error is not thrown synchronously, but asynchronously.